docs/proposals/compatibility-mode.md

```diff
@@ -0,0 +1,60 @@
+# Proposal: Provide a compatibility mode for ORAS
+
+This document is for adding a proposal to enable compatibility mode for ORAS CLI and solve [issue #720](https://github.com/oras-project/oras/issues/720).
+
+## Background
+
+OCI group announced the release of v1.1 for [Image-spec](https://github.com/opencontainers/image-spec/blob/main/artifact.md) and [Distribution-spec](https://github.com/opencontainers/distribution-spec) in Sep 2022. It supports the OCI artifact manifest and provides a new referrers discovery API that allows artifact references.
+
+Since 0.16.0, ORAS supports pushing OCI artifact manifest to OCI v1.0 compliant registries. However, the new manifest type may not be supported on the consumer side (e.g. self-crafted scripts) or in those OCI v1.0 compliant registries. To enable ORAS to work with popular registries, it provides backward compatibility which supports two types of manifest and allows fallback to upload the OCI image manifest to OCI v1.0 compliant registries or enabled OCI image manifest storage. 
+
+## Challenge
+
+However, the ORAS fallback may fail as there is no deterministic way to confirm if a registry supporting OCI artifact manifest and no consistent error response. 
+You can find the testing result of the implementation result for OCI Spec from this [blog](https://toddysm.com/2023/01/05/oci-artifct-manifests-oci-referrers-api-and-their-support-across-registries-part-1/). On the other hand, users may want to force-push a specific manifest type to a registry for some reason.
+
+The current workaround for enabling a kind of compatibility mode is to specify a `--config` flag when using `oras push`. Since the OCI artifact manifest does not have a `config`, it will push an OCI image manifest instead. It is not user-friendly and is a bit hacky. It would be better if we can provide a compatibility mode to easily customize and switch the manifest uploading behavior, and enable users to handle the incompatibility problem when using ORAS across different registries. 
+
+## Goals
+
+- Enable ORAS to work with more registries
+- Provide different options to allow users to customize the behaviors of uploading manifest to the registry
+- Provide an easy-to-use and secure user experience when switching the behaviors
+
+## Solution
+
+Adding a new flag `--compatibility` under CLI commands `oras push` and `oras attach` with different variables to configure the behaviors of uploading manifest. We will only use `oras push` to demonstrate the examples below.
```

Comment:
I have concerns on the forward compatibility. What if we have a new manifest in the future? In that case, `min` is no longer `min`.

How about using specific terms instead of `min` and `max`? or even introduce something like [compatibility level](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver16#supported-dbcompats)?

Reply:
Proposed terms are
- `min` --> `v1.1-referrers`
- `max` --> `v1.1-fallback`
- `artifact-manifest` --> `v1.1`, `v1.1-artifact`
- `image-manifest` --> `v1.1-image`

Additionally, the default one is named `auto`.

Reply:
Based on the proposed terms, why not just name `--compatibility` as `--spec`? Much simpler 😺